### PR TITLE
Use ReturnHost value returned by API to compare origin

### PR DESCRIPTION
**Pull request by EpokK**
Use `ReturnHost` value returned by API to compare origin.
```json
{
    "Code": 1000,
    "Token": "mLN7SpUGz25ol1X5wDoDGOGG",
    "Status": 0,
    "ApprovalURL": "https://hooks.stripe.com/3d_secure_2/hosted?publishable_key=pk_test_FgIybmWSP4xgpBTsUJMFYG1k&setup_intent=seti_1FNGJZKpW9DKy5hnIiHwWvlZ&setup_intent_client_secret=seti_1FNGJZKpW9DKy5hnIiHwWvlZ_secret_Ft2OrbowBdg1WtE7aOOJuRxuMiuh00c&source=src_1FNGJZKpW9DKy5hn7lwfDVIq",
    "ReturnHost": "protonmail.blue/api"
}
```

### CODE REVIEW

File: containers/payments/PayPal.js [hunk 1]
```diff
@@ -13,9 +13,9 @@ const PayPal = ({ amount: Amount, currency: Currency, onPay, type }) => {
     const [loadingToken, withLoadingToken] = useLoading();
     const [loadingVerification, withLoadingVerification] = useLoading();
     const [textError, setTextError] = useState('');
-    const [approvalURL, setApprovalURL] = useState();
-    const [token, setToken] = useState();
-    const { SECURE_URL: secureURL } = useConfig();
+    const [ApprovalURL, setApprovalURL] = useState();
+    const [Token, setToken] = useState();
+    const [ReturnHost, setReturnHost] = useState();
```
Comment (mmso) [inline containers/payments/PayPal.js]: Perhaps these 3 would be better suited in a ref

File: containers/payments/paymentTokenHelper.js [hunk 2]
```diff
@@ -101,7 +101,7 @@ export const process = ({ Token, api, approvalURL, secureURL, signal }) => {
         const onMessage = (event) => {
             const origin = event.origin || event.originalEvent.origin; // For Chrome, the origin property is in the event.originalEvent object.
 
-            if (origin !== secureURL) {
+            if (!origin.includes(ReturnHost)) {
```
Comment (mmso) [inline containers/payments/paymentTokenHelper.js]: Did you use `includes` because the `ReturnHost` does not contain `https://`?

Comment (EpokK) [reply containers/payments/paymentTokenHelper.js]: I changed the logic.

**Comment by EpokK** [comment]
API change is not yet deployed on live, @N4M3Z will inform us when it is.

**Pull request merged**